rfd/0194-k8s-rbac-simplification.md

```diff
@@ -0,0 +1,220 @@
+role `system:masters`, everything works as expected. Teleport is super admin
+and yields reduced permissions to the user based on the `kubernetes_resources`
+and `kubernetes_label` fields from the Teleport role.
+
+### Impersonation
+
+To limit the number of code changes, the proposal is to default the
+`kubernetes_group` value to `system:masters` and hide it from the role.
+This can be done by allowing the field to be `null`, and treating `null` as
+`systems:masters`.
+
+The `kubernetes_user` should default to the Teleport user auth email.
+
+### Labels
+
+We currently support labels at the cluster level, however, resources only
+support name/namespace filtering.
+Adding an optional field for labels would allow for more flexible and versatile
+control over permissions, allowing something like:
+
+```yaml
+kubernetes_resources:
+  - kind: '*'
+    name: '*'
+    namespace: '*'
+    verbs: ['*']
+    labels:
+      region: europe
+      stage: development
+```
+
+This would grant access to any resource with the matching labels `region` and
+`stage`. The benefit is that it simplifies a lot granting access to many
+resources. Granting all of a namespace can be too permissive and granting
+individually `ingress`, `deployments`, `services`, `pods`, etc, etc can be
+cumbersome and error prone.
+
+### UX
+
+#### Deprecate `namespace`
+
+We currently have the `namespace` kind mean "everything within a namespace",
+which has caused confusion with customers and resulting in unexpected access
+granted.
+
+This should be deprecated and removed. It would require a bump on the role
+model to v8 to avoid changing behavior for current customers.
+
+Upgrading to v8 would map the deprecated `namespace` field to `*` while
+swapping `name`/`namespace` as these are equivalent:
+
+```yaml
+kind: namespace
+name: foo
+```
+
+and:
+
+```yaml
+kind: '*'
+namespace: foo
+```
+
+#### Web Role Editor
+
+To avoid confusion, in the web role editor, we would remove the ability to set
+and view `kubernetes_group` and `kubernetes_users` in favor of only allowing it
+to be controlled in yaml.
+When set in an existing role, similar to the existing `access` role, the UI can
+be disabled with a message saying:
+
+```text
+This role is too complex to be edited in the standard editor. To continue
+editing, go back to YAML editor.
+```
+
+The `kubernetes_resources` section currently "hidden" behind a button should be
+open by default.
+
+Alternatively, we could keep `kubernetes_group` under an "advanced" button,
+which would show what we see today by default as well as a link to a specific
+documentation page.
+
+#### CLI
+
+Using `tctl`, the fields will only be visible if they are set. Existing roles
+will have the existing values, even if already set to `kubernetes_master`,
+however, new roles can be created without the field and will be surfaced back
+without the field either.
+
+#### Direct enrollment
+
+When enrolling a cluster "directly" (i.e. by adding the `kubernetes_service`
```

Comment:
For direct enrollment we can ask users to grant `cluster-admin` to Teleport service account